test/integration/acceptance/queries_acceptance_test.cpp

```diff
@@ -0,0 +1,487 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "backend/protobuf/transaction.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+#include "interfaces/permissions.hpp"
+#include "utils/query_error_response_visitor.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class QueriesAcceptanceTest : public AcceptanceFixture {
+ public:
+  QueriesAcceptanceTest()
+      : itf(1),
+        invalidPrivateKey(kUserKeypair.privateKey().hex()),
+        invalidPublicKey(kUserKeypair.publicKey().hex()) {
+    /*
+     * It's deliberately get broken the public key and privite key to simulate a
+     * non-valid signature and public key and use their combinations in the
+     * tests below
+     * Both keys are hex values represented as a std::string so
+     * characters can be values only in the range 0-9 and a-f
+     */
+    invalidPrivateKey[0] == '9' or invalidPrivateKey[0] == 'f'
+        ? --invalidPrivateKey[0]
+        : ++invalidPrivateKey[0];
+    invalidPublicKey[0] == '9' or invalidPublicKey[0] == 'f'
+        ? --invalidPublicKey[0]
+        : ++invalidPublicKey[0];
+  };
+  void SetUp() {
+    itf.setInitialState(kAdminKeypair)
+        .sendTx(makeUserWithPerms({interface::permissions::Role::kGetRoles}))
+        .skipProposal()
+        .checkBlock([](auto &block) {
+          ASSERT_EQ(boost::size(block->transactions()), 1);
+        });
+  };
+
+  template<typename F>
+  auto checkIfStatefulFailed(F& queryResponse)
```

Comment:
All three verification functions do not access any class members of the fixture.
Making them static will let you eliminate usage of such proxy lambdas:
```
auto check = [this](auto &queryResponse){checkIfStatefulFailed(queryResponse);};
```

Please make them static and remove proxy lambdas.
The usage would be quite simple:
```
itf.sendQuery(query, checkIfStatefulFailed);
```